akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolConductor.scala

```diff
@@ -153,6 +164,27 @@ private object PoolConductor {
       override def preStart(): Unit = {
         pull(ctxIn)
         pull(slotIn)
+
+        startMinConnections()
+      }
+
+      def startMinConnections(): Unit =
+        (0 until slotSettings.minSlots).foreach { connect }
+
+      // pool ordering connection doesn't change the status of a connection to Idle,
+      // receiving confirmation from the Slot does. See: handler for "slotIn"
+      def connect(index: Int): Unit =
+        emit(out, SlotShouldConnectCommand(index))
+
+      private def reconnectIfNeeded(): Unit = {
+        val connectedNum = slotStates.count(_ != Unconnected)
+        if (connectedNum < slotSettings.minSlots) {
+          val unconnected = slotStates
+            .zipWithIndex
+            .filter(_._1 == Unconnected)
+            .take(slotSettings.minSlots - connectedNum)
+          unconnected.map(_._2).map(connect)
+        }
```

Comment:
I will double-check if `reconnectIfNeeded()` should only always connect one slot. If it's the case I will go with`indexWhere` as you suggested. Thanks